platform-api/src/main/java/io/datacater/core/deployment/DeploymentSpec.java

```diff
@@ -28,4 +32,17 @@ public record DeploymentSpec(
   public DeploymentSpec(String name, Map<String, Object> deployment) {
     this(name, deployment, new HashMap<>());
   }
+
+  public static DeploymentSpec from(DeploymentSpec spec) {
+    Map<String, Object> copiedMap = new HashMap<>();
+    try {
+      JsonNode specNode = DeploymentEntity.serializeMap(spec.deployment());
+      ObjectMapper mapper = new ObjectMapper();
+      copiedMap = mapper.treeToValue(specNode, Map.class);
+    } catch (JsonProcessingException e) {
+      throw new DatacaterException(e.getMessage());
```

Comment:
Is the "mapping" to a `DataCaterException` needed for returning a meaningful response to users in the case of an error?